programs/mp-sol-restaking/src/instructions/cranks/update_attached_strat_lst_amount.rs

```diff
@@ -92,8 +92,9 @@ pub struct UpdateAttachedStratLstAmount<'info> {
     )]
     pub mpsol_mint_authority: UncheckedAccount<'info>,
 
-    #[account(mut, token::mint = mpsol_mint)]
-    pub treasury_mpsol_account: Account<'info, TokenAccount>,
+    #[account(mut)]
```

Comment:
Please make it optional so we can skip providing it if the field is None and add a check to the code (not macro) if it is required and set to the correct address but not a token account just ignore. If it is not ATA it may be recreated as non-token account by its owner. This must be treated the same way as None (admin don't wanna get fees)
Plus add a fee_amount field into the event struct

Reply:
1) we need the treasury account to get the withdrawal fee, so it should never be None
2) Will add the check if it is present but not a token account to avoid admin locking withdrawal
3) adding to event